WalletWasabi/Services/HostedServices.cs

```diff
@@ -16,6 +17,8 @@ public class HostedServices : IDisposable
 	private object ServicesLock { get; } = new();
 	private bool IsStartAllAsyncStarted { get; set; } = false;
 
+	public event EventHandler<IHostedService>? ServiceRegistered;
```

Comment:
I had the same brainstorm myself. That would be the most idealistic. But Global was built in a way because there are many service that needs time to init and in some cases, only after that do you decide to create another service, you might not need it at all. For example RpcMonitor needs IRPCClient which needs CoreNode. So would you construct RpcMonitor without a component that is fundamental for it and register later IRPCClient? - "it is not good" - author.

Reply:
I think it is a legit case for the UI to wait for a service that is not ready or not exist and hook up to it.